src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyConsoleLogger.cs

```diff
@@ -2,12 +2,50 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
 using System.Text;
 using Microsoft.Extensions.Logging;
 using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.WebAssembly.Services
 {
+    /// <summary>
+    /// A provider of <see cref="WebAssemblyConsoleLogger{T}"/> instances.
+    /// </summary>
+    public class WebAssemblyConsoleLoggerProvider : ILoggerProvider
+    {
+        private ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>> _loggers;
```

Comment:
This is perfectly fine. If you wanted to, you could probably get away with using a vanilla `Dictionary` since the environment is single threaded.

Reply:
```suggestion
        private readonly ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>> _loggers;
```

Reply:
@pranavkm We're typically trying to account for thread safety in the usual multi-threaded style now whenever possible, because pretty soon we'll want to enable the threaded version of Mono WebAssembly. At that point we'll need to go through the code and hunt down any thread safety issues due to older assumptions.

Reply:
Oh fun. That makes `WebAssemblyConfigurationBuilder : IConfigurationBuilder, IConfiguration` a tad bit trickier.